### PR TITLE
WIP: add docker validation test

### PR DESCRIPTION
**Pull request by liangchenye**
The docker validation test including:
1. conformance test
2. performance test
3. docker unique test (restart docker on upgrade for example)
Since the docker unique testing affects other container test, for example, list/status.
I think we cannot test them simultaneously in docker validation test.
So I'll test 1~3 one by one.

[conformance test]
In current e2e-node test, we already have very solid code to setup k8s environment.
I think we can just reuse them, but will be more parameters (like test on a certain GCE?)
(`docker-validation-test.sh` is modifies from e2e-node-test.sh)

[performance test]
I'm not sure how to integrate with @Random-Liu 's work .

[docker unique test]
I add a 'restart' testcase in this PR. It works good in my docker version..

@dchen1107 @Random-Liu

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/25471)

### CODE REVIEW

**Comment by Random-Liu** [comment]
@liangchenye Nice!

File: test/docker_validation/docker.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	err := cmd.Start()
+	if sync {
+		err = cmd.Wait()
+	}
+
+	return err
+}
+
+func (cc *ConformanceContainer) Start() error {
+	_, err := exec.Command("service", cc.Name, "start").CombinedOutput()
```
Comment (dchen1107) [inline test/docker_validation/docker.go]: This won't work on systemd node, right?

Comment (liangchenye) [inline test/docker_validation/docker.go]: Yes, I will add more checking codes to the 'start' function.

**Comment by liangchenye** [comment]
@Random-Liu I add your `helpers` code and one of your case.
Just as we discussed today, I'll submit small case(code) first.

@dchen1107 I update previous limitation code by supporting both 'init/systemd'.
(get reference from https://github.com/autotest/autotest-docker/blob/master/dockertest/docker_daemon.py )

I'll remove the 'WIP' and ask for a review once the jenkins integration is done.

**Comment by liangchenye** [comment]
@Random-Liu  I add 4 cases: start/start-in-parallel/list/list-in-parallel.
Each measurement has 10 containers and runs related operation twice.
The output is like this: 
```
• [MEASUREMENT]
start:
  Fastest Time: 3.920s
  Slowest Time: 3.936s
  Average Time: 3.928s ± 0.008s

start parallelly:
  Fastest Time: 0.692s
  Slowest Time: 0.886s
  Average Time: 0.789s ± 0.097s

list:
  Fastest Time: 0.154s
  Slowest Time: 0.158s
  Average Time: 0.156s ± 0.002s

list parallelly:
  Fastest Time: 0.132s
  Slowest Time: 0.134s
  Average Time: 0.133s ± 0.001s
```

**Comment by Random-Liu** [comment]
@k8s-bot test this issue #IGNORE.

**Comment by Random-Liu** [comment]
`FAILED   ./hack/../hack/verify-boilerplate.sh	1s`

**Comment by dchen1107** [comment]
ping ... 

@liangchenye any updates on this pr?

**Comment by dchen1107** [comment]
ping ...

**Pull request closed without merging**